contrib/terraform/common/common.tf

```diff
@@ -0,0 +1,93 @@
+variable "temp_basedir" {
+  type    = "string"
+  default = "/tmp"
+}
+
+variable "inventory_deploy" {
+  default = true
+}
+
+resource "random_string" "inventory_dir" {
+  length = 10
+  special = false
+}
+
+locals {
+  inventory_file = "${var.temp_basedir}/terraform-${random_string.inventory_dir.result}/hosts.ini"
+}
+
+output "inventory_file" {
+  value = "${local.inventory_file}"
+}
+
+variable "ssh_key_path" {
+  type    = "string"
+  default = "~/.ssh/id_rsa.pub"
+}
+
+variable "ssh_key_name" {
+  type    = "string"
+  default = "metalk8s"
+}
+
+variable "ssh_key_deploy" {
+  default = true
+}
+
+output "ssh_key_path" {
+  value = "${var.ssh_key_path}"
+}
+
+variable "volume_size" {
+  type    = "string"
+  default = 200
+}
+
+output "volume_size" {
+  value = "${var.volume_size}"
+}
+
+variable "nodes_count" {
+  type    = "string"
+  default = 5
+}
+
+variable "etcd_count" {
+  type    = "string"
+  default = 5
+}
+
+variable "etcd_dedicated" {
+  default = true
+}
+
+variable "masters_count" {
+  type    = "string"
+  default = 3
+}
+
+variable "masters_dedicated" {
```

Comment:
Is there a way to colocate etcd and master (control plane), not nodes?

Reply:
there is both:
masters_dedicated (default=false) => by default, masters are colocated on etcd nodes
etcd_dedicated (default=true) => by default, etcd spawn dedicated VM , if false etcd join nodes

which result in three case:
- everything colocated on nodes ( -var "etcd_dedicated=0")
- nodes one group, etcd and masters the other group (the default)
- everything gets it's own group (-var "masters_dedicated=1)

Reply:
Suggestion on what's the best API to do that are welcome